controller/eks-cluster-config-handler.go

```diff
@@ -1020,21 +1062,47 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 		// happen together
 
 		ng := ngs[aws.StringValue(upstreamNg.NodegroupName)]
-		if ng.Version != nil {
-			if aws.StringValue(upstreamNg.Version) != desiredNgVersions[aws.StringValue(ng.NodegroupName)] {
-				attemptUpdatingNodegroups = true
-				_, err := eksService.UpdateNodegroupVersion(
-					&eks.UpdateNodegroupVersionInput{
-						NodegroupName: aws.String(aws.StringValue(ng.NodegroupName)),
-						ClusterName:   aws.String(config.Spec.DisplayName),
-						Version:       aws.String(desiredNgVersions[aws.StringValue(ng.NodegroupName)]),
-					},
-				)
+		ngVersionInput := &eks.UpdateNodegroupVersionInput{
+			NodegroupName: aws.String(aws.StringValue(ng.NodegroupName)),
+			ClusterName:   aws.String(config.Spec.DisplayName),
+		}
+
+		if upstreamNg.LaunchTemplate != nil {
+			upstreamTemplateVersion := aws.Int64Value(upstreamNg.LaunchTemplate.Version)
+			var err error
+			lt := ng.LaunchTemplate
+
+			if lt == nil && config.Status.LaunchTemplateID == aws.StringValue(upstreamNg.LaunchTemplate.ID) {
+				// In this case, Rancher is managing the launch template, so we check to see if we need a new version.
+				lt, err = newLaunchTemplateVersionIfNeeded(config, upstreamNg, ng, ec2Service)
 				if err != nil {
 					return config, err
 				}
-				continue
 			}
+
+			if lt != nil && (aws.StringValue(lt.ID) != aws.StringValue(upstreamNg.LaunchTemplate.ID) || aws.Int64Value(lt.Version) != upstreamTemplateVersion) {
```

Comment:
I thought the launch template ID could not be change on a nodegroup, is that incorrect?

Reply:
You can change the launch template ID in EKS.

Reply:
Ah, so syncing could cause it. Do we allow users to update the template ID in Rancher?